### ISSUE TITLE
Module Inclusion Suggestion: Search_API_exclude

### ISSUE THREAD

**Issue by tobybellwood** [govCMS/GovCMS7#495]
It has been discussed in a support ticket that the **Search_API_exclude** module (https://www.drupal.org/project/search_api_exclude) could be considered for inclusion in the distribution.

This would provide the ability to exclude nodes from indexing by the Search API, in addition to the existing ability to exclude entire content types.

Thoughts?  Is this something with wider demand or benefit? Happy to have the discussion!

There is a corresponding discussion on the community site for those who don't GitHub at https://community.govcms.gov.au/t/module-inclusion-suggestion-search-api-exclude

**Comment by nathan-w** [comment]
Id be keen to know a bit more about the actual business need. Is it excluding content from Drupal search, how would it affect our use of SOLR? What about Google Search? 

If the content is still discoverable and there are no robots rules excluding the indexing, the content is still likely to be discovered on Google Search anyway.... so what does this actually achieve?

**Comment by dman-coders** [comment]
Yeah, there is a limit to what Search_API_exclude can do you are talking about Google explicitly, but as far as Solr + Search_api, that's push-driven (in the common setup) so excluding it from search_api would be somewhat effective.

From what it says on the box, I would have hoped that Search_API_exclude would also have tried to add some meta no-index type headers to anticipate the Google question. Sadly, it does not. It looks like a pretty weak module altogether from code review.
Maybe it's for this reason that I saw an alternative: https://www.drupal.org/project/search_exclude_nid
- this also may not pass code review as it does not scale well, but it does describe the use-case for this problem.

Basically, sometimes you end up making a site out of chunks of content that don't really need to be accessed on their own or out of context. Sometimes (especially after a client goes looking for their own specific keywords) it's nice to have those pages less obvious.

A more effective way to make second-class content un-findable (depending on use-case) for us has been https://www.drupal.org/project/rabbit_hole

**Comment by rooby** [comment]
Just saw this issue, but I also opened #498 for rabbit hole.

Without knowing more about the OPs requirements I would say Search_API_exclude type functionality is likely not the best way to solve most problems.
